### PR TITLE
removing device bindings in HYPRE_Init

### PR DESCRIPTION
**Pull request by liruipeng**
This PR removes GPU device binding in `HYPRE_Init()`.

### CODE REVIEW

File: src/utilities/hypre_cuda_utils.c [hunk 1]
```diff
@@ -1077,3 +1077,51 @@ hypre_SyncCudaComputeStream(hypre_Handle *hypre_handle)
...
+   hypre_MPI_Comm_size(node_comm, &NodeSize);
+   hypre_MPI_Comm_free(&node_comm);
+
+   /* get number of devices on this node */
+#if defined(HYPRE_USING_CUDA)
+   HYPRE_CUDA_CALL( cudaGetDeviceCount(&nDevices) );
+#elif defined(HYPRE_USING_DEVICE_OPENMP)
+   nDevices = omp_get_num_devices();
+#else
+#error hip
```
Comment (liruipeng) [inline src/utilities/hypre_cuda_utils.c]: @pbauman Please help fill this gap, i.e., HIP's `GetDeviceCount`. Thanks!

File: src/utilities/hypre_cuda_utils.c [hunk 2]
```diff
@@ -1077,3 +1077,51 @@ hypre_SyncCudaComputeStream(hypre_Handle *hypre_handle)
...
+
+   /* set device */
+   device_id = myNodeid % nDevices;
+#if defined(HYPRE_USING_CUDA)
+   HYPRE_CUDA_CALL( cudaSetDevice(device_id) );
+#elif defined(HYPRE_USING_DEVICE_OPENMP)
+   HYPRE_CUDA_CALL( cudaSetDevice(device_id) );
+   omp_set_default_device(device_id);
+#else
+#error hip
```
Comment (liruipeng) [inline src/utilities/hypre_cuda_utils.c]: @pbauman Please help fill this gap, i.e., HIP's `SetDevice`. Thanks!

File: src/utilities/hypre_memory.c [hunk 3]
```diff
@@ -1248,9 +1248,13 @@ HYPRE_Int
 hypre_umpire_device_pooled_allocate(void **ptr, size_t nbytes)
 {
    hypre_Handle *handle = hypre_handle();
+   //const hypre_int device_id = hypre_HandleCudaDevice(handle);
```
Comment (liruipeng) [inline src/utilities/hypre_memory.c]: @Ram3sh @davidbeckingsale I tried to add Umpire support for multiple GPUs following instructions from @Ram3sh with `DEVICE::X` and `UM::X` where `X` is device id (see commented code here), but got segfault. Any ideas of what I am doing wrong? Thanks!

Comment (davidbeckingsale) [reply src/utilities/hypre_memory.c]: What version of Umpire are you using?

Comment (liruipeng) [reply src/utilities/hypre_memory.c]: @davidbeckingsale

I use the `develop` branch up to this commit.

```
commit 6ba5e4dd9756cb0cbc4a204443bc96c075d587d3 (HEAD -> develop, origin/develop, origin/HEAD)
Merge: 0054f38f 8180fabd
Author: David Beckingsale <davidbeckingsale@gmail.com>
Date:   Tue Jan 12 16:01:24 2021 -0800

    Merge pull request #521 from LLNL/bugfix/UM-815-fix-CMakeLists
    
    removing extra print statement in CMakeLists
```

Comment (liruipeng) [reply src/utilities/hypre_memory.c]: Hi, @davidbeckingsale, is this problem easy to fix? If not, we may leave it as is and merge the other parts of this PR in. Just asking. Thanks! -Ruipeng

Comment (davidbeckingsale) [reply src/utilities/hypre_memory.c]: Everything looks fine - can you email me more details about the segfault?

**Comment by liruipeng** [comment]
@Ram3sh @rfalgout @ulrikeyang @oseikuffuor1 This PR is good to go on my end. Please take a look. Thanks!

**Review by rfalgout — APPROVED** [review]
Looks good to me.  Thanks!

**Review by oseikuffuor1 — APPROVED** [review]
Thanks

**Review by Ram3sh — APPROVED** [review]
Looks good to me

**Review by ulrikeyang — APPROVED** [review]

**Pull request merged**